src/state/wallet/hooks.ts

```diff
@@ -1,98 +1,76 @@
 import {
   BLOCK_TIME,
   DAI,
-  STABLECOIN_POOL_NAME,
-  Token,
-  VETH2,
-  VETH2_POOL_NAME,
-  WETH,
-} from "../../constants"
-import {
-  BTC_POOL_NAME,
-  PoolName,
   RENBTC,
   SBTC,
   TBTC,
   USDC,
   USDT,
+  VETH2,
   WBTC,
+  WETH,
 } from "../../constants"
+import { Contract, Provider } from "ethcall"
 
 import { BigNumber } from "@ethersproject/bignumber"
-import { Erc20 } from "../../../types/ethers-contracts/Erc20"
-import { Zero } from "@ethersproject/constants"
+import { Call } from "ethcall/lib/call"
+import ERC20_ABI from "../../constants/abis/erc20.json"
+import { IS_DEVELOPMENT } from "../../utils/environment"
+import { getNetworkLibrary } from "../../connectors"
 import { useActiveWeb3React } from "../../hooks"
-import { useMemo } from "react"
 import usePoller from "../../hooks/usePoller"
 import { useState } from "react"
-import { useTokenContract } from "../../hooks/useContract"
 
-export function useTokenBalance(t: Token): BigNumber {
+export function usePoolTokenBalances(): { [token: string]: BigNumber } | null {
   const { account, chainId } = useActiveWeb3React()
-  const [balance, setBalance] = useState<BigNumber>(Zero)
+  const [balances, setBalances] = useState<{ [token: string]: BigNumber }>({})
 
-  const tokenContract = useTokenContract(t) as Erc20
+  const ethcallProvider = new Provider()
 
   usePoller((): void => {
-    async function pollBalance(): Promise<void> {
-      const newBalance = account
-        ? await tokenContract?.balanceOf(account)
-        : Zero
-      if (newBalance !== balance) {
-        setBalance(newBalance)
+    async function pollBalances(): Promise<void> {
+      if (!chainId) return
+
+      await ethcallProvider.init(getNetworkLibrary())
```

Comment:
I think you should be able to use `const {library} = useActiveWeb3React()` here instead